src/main/java/com/openshift/restclient/capability/resources/IPropertyAccessCapability.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.openshift.restclient.capability.resources;
+
+import java.util.Map;
+
+import com.openshift.restclient.capability.ICapability;
+
+/**
+ * A mechanism to access the underlying content of a json
+ * structure
+ * 
+ * Relies on a dot delimited path to indentify the root of the entity
+ * being returned (e.g. 'metadata.labels').  This does not provide
+ * a mechanism to retrieve properties across an array kind.
```

Comment:
I dont see it as a big deal since you are able to retrieve the entire JSON as a map of string to Objects.  But the bigger question is what does it mean to retrieve an entry across an array?  Do we allow you to do something like "foo.[3].bar" where you retrieve bar value of the third entry?  I think this is probably good enough but if we can come up with a could API, I'm willing to adjust it.